File: WCF/Windows SDK WCF Samples/TechnologySamples Beta 1/Scenario/LoadBalancer/CS/library/ILoadBalancer.cs

//  Copyright (c) Microsoft Corporation.  All Rights Reserved.

using System;
using System.Xml;
using System.ServiceModel;

namespace Microsoft.ServiceModel.Samples
{
    [ServiceContract(Namespace = "http://Microsoft.ServiceModel.Samples")]
    public interface ILoadBalancer
    {
        [OperationContract]
        string Register(string contractQName, string bindingQName, EndpointAddress10 serviceAddress);

        [OperationContract(IsOneWay = true)]
        void Unregister(string registrationId);

        [OperationContract]
        EndpointAddress10 GetServiceAddress(string contractQName, string bindingQName);
    }
}

File: WCF/Windows SDK WCF Samples/TechnologySamples Beta 1/Scenario/LoadBalancer/VB/library/ILoadBalancer.vb
' Copyright (c) Microsoft Corporation.  All Rights Reserved.

Imports System
Imports System.Xml
Imports System.ServiceModel

Namespace Microsoft.ServiceModel.Samples

    <ServiceContract([Namespace]:="http://Microsoft.ServiceModel.Samples")> _
    Public Interface ILoadBalancer

        <OperationContract()> _
        Function Register(ByVal contractQName As String, ByVal bindingQName As String, ByVal serviceAddress As EndpointAddress10) As String
        <OperationContract(IsOneWay:=True)> _
        Sub Unregister(ByVal registrationId As String)
        <OperationContract()> _
        Function GetServiceAddress(ByVal contractQName As String, ByVal bindingQName As String) As EndpointAddress10

    End Interface

End Namespace
